### PR TITLE
fix #21

### LINKED ISSUE #21: InstructionReport.merge method needs adjustments
object's address should not be taken into equality comparison

 Q{aaa:/path/module.xq}name($param, ``com.saxonica.functions.hof.CallableFunctionItem@6bee793f)`` <> Q{aaa:/path/module.xq}name($param, ``com.saxonica.functions.hof.CallableFunctionItem@7942a854)``

### PR DESCRIPTION
**Pull request by mjedynak**

### CODE REVIEW

File: coverage/src/main/java/com/github/lizardev/xquery/saxon/coverage/trace/CoverageExpression.java [hunk 1]
```diff
@@ -20,7 +20,16 @@ public CoverageExpression(Expression child, CoverageInstructionEventHandler even
 
     @Override
     public String toString() {
-        return getChild().toString();
+        String childToString = getChild().toString();
+        if (childToString.contains("@")) {
```
Comment (alaczynski) [inline coverage/src/main/java/com/github/lizardev/xquery/saxon/coverage/trace/CoverageExpression.java]: what about xpaths which contain @? as example Functx-1.0-nodoc-2007-01.xq line 424

File: coverage/src/test/java/com/github/lizardev/xquery/saxon/coverage/trace/CoverageExpressionTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+@RunWith(MockitoJUnitRunner.class)
+public class CoverageExpressionTest {
+
+    @Mock private Expression expression;
+    @Mock private CoverageInstructionEventHandler handler;
+
+    @Test
+    public void shouldRemoveObjectsAddressInToString() {
+        CoverageExpression coverageExpression = new CoverageExpression(new TraceExpression(expression) {
```
Comment (alaczynski) [inline coverage/src/test/java/com/github/lizardev/xquery/saxon/coverage/trace/CoverageExpressionTest.java]: no sure if TraceExpression is needed as additional wrapper, in our code we have only CoverageExpression(expresion).

File: coverage/src/test/java/com/github/lizardev/xquery/saxon/coverage/trace/CoverageExpressionTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        String result = coverageExpression.toString();
+        
+        assertThat(result).isEqualTo("com.github.lizardev.xquery.saxon.coverage.trace.CoverageExpressionTest$ClassWithNoToStringImplementation");
+    }
+
+    @Test
+    public void shouldNotTryToRemoveObjectsAddressInToStringIfItIsAbsent() {
+        String stringRepresentation = "stringRepresentation";
+        given(expression.toString()).willReturn(stringRepresentation);
+        CoverageExpression coverageExpression = new CoverageExpression(new TraceExpression(expression), handler);
```
Comment (alaczynski) [inline coverage/src/test/java/com/github/lizardev/xquery/saxon/coverage/trace/CoverageExpressionTest.java]: same as above

File: coverage/src/main/java/com/github/lizardev/xquery/saxon/coverage/trace/CoverageExpression.java (hunk 1 above)
Comment (mjedynak) [inline coverage/src/main/java/com/github/lizardev/xquery/saxon/coverage/trace/CoverageExpression.java]: I'd go with some simpler solution (than the list) first.

**Pull request closed without merging**